external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaOffsetReader.scala

```diff
@@ -105,34 +109,16 @@ private[kafka010] class KafkaOffsetReader(
     minPartitions.map(_ > numTopicPartitions).getOrElse(false)
   }
 
-  private def nextGroupId(): String = {
-    groupId = driverGroupIdPrefix + "-" + nextId
-    nextId += 1
-    groupId
-  }
-
   override def toString(): String = consumerStrategy.toString
 
   /**
    * Closes the connection to Kafka, and cleans up state.
    */
   def close(): Unit = {
-    if (_consumer != null) uninterruptibleThreadRunner.runUninterruptibly { stopConsumer() }
+    if (_admin != null) uninterruptibleThreadRunner.runUninterruptibly { stopAdmin() }
     uninterruptibleThreadRunner.shutdown()
   }
 
-  /**
-   * @return The Set of TopicPartitions for a given topic
-   */
-  def fetchTopicPartitions(): Set[TopicPartition] = uninterruptibleThreadRunner.runUninterruptibly {
-    assert(Thread.currentThread().isInstanceOf[UninterruptibleThread])
```

Comment:
I'm OK to keep the check, but in the following commit I expect we get rid of using UninterruptibleThread altogether.

https://github.com/apache/spark/pull/29729/files#r488436868

Reply:
I've just finished the testing of `UninterruptibleThread` removal from `KafkaOffsetReader`. We can put it here if we agree. Both side we can come up w/ pros and cons.

> get rid of using UninterruptibleThread altogether.

Not sure it's possible or at least not super easy. In executor side it's used because of multiple reasons. Please see: https://github.com/apache/spark/blob/2e3aa2f0232a539346da3df8a20cd8e7c2b7dd4f/core/src/main/scala/org/apache/spark/executor/Executor.scala#L101-L104

Reply:
My rationale related removing here vs removing in separate PR is the following: I personally don't see a reason why the removal could cause any issues but to be defensive we can put it into a separate PR. This way if any issue appears for mystical reason we can rollback that PR independently. Does this make sense?

Reply:
Sorry I meant Kafka related, driver side only.

Reply:
Having a separate PR is OK for me - that would avoid unnecessary hold on this PR.

Reply:
I agree w/ this way.